### PR TITLE
Please review my PR and suggest me if need any changes

### PR DESCRIPTION
**Pull request by crooknish**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,140 @@
 # Smartwatch
-Open Projects 2021
+Open Projects 2021
```
Comment (theobscuredev) [inline README.md]: Apply H2 heading to this text using '## Open Projects 2021'

File: README.md [hunk 2]
```diff
@@ -1,2 +1,140 @@
 # Smartwatch
-Open Projects 2021
+Open Projects 2021  
+
+## Abstract  
+This smartwatch project largely deals with the exploration of what is Arduino and how it works. A Smartwatch functions both as a watch as well as a miniature size smart phone with limited features. Designing and simulating an Arduino based smartwatch requires a lot of knowledge about the hardware part. But this project was entirely designed and simulated in online mode using various softwares to draw the schematic and simulate it (Proteus), work on the PCB layout (Easy EDA) and finally do a casing for the smartwatch (solidworks).   
+
+This watch can be connected to mobile via Bluetooth and it can respond to the commands given via the mobile too. It can also monitor the notifications that we receive in our mobile and alert the user accordingly. Apart from this, it can also show the day-date-time information like a normal digital watch and can also show us the temperature of the medium in which it is kept.
```
Comment (theobscuredev) [inline README.md]: Justify the entire text using the `<p>` paragraph tag and `justify` attribute. Make sure you justify all the paragraphs separately. For eg:
`<p align="justify">`
  `<h2>Abstract</h2>`
  `<p>A Rube Goldberg Machine is .........................</p>`
`</p>`

Refer this README for the same: https://raw.githubusercontent.com/marsiitr/Rube-Goldberg-Machine/master/README.md

File: README.md [hunk 3]
```diff
@@ -1,2 +1,140 @@
...
+  
+<p align="center">
+  <img src="https://github.com/crooknish/Smartwatch/blob/b9872868b3e2b396aa44f89e3450bc2ef973ca1a/Images%20and%20Videos/PCB%203D%20view.jpeg"></br>
+  <i> PCB 3D Model </i>
+  </p>
```
Comment (theobscuredev) [inline README.md]: Insert this image above your Abstract as specified in the guidelines.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,140 @@
...
+  
+
+## Motivation
+This project mainly aimed to create a rough model of a smartwatch that will resemble a typical smartwatch in functioning but, we however optimized the design so that it will also resemble a typical smartwatch in appearance too. However, the ultimate motive of this project was to learn the coding of the Arduino and testing it along with different sensors and modules that are connected to the Arduino. So, we were least bothered about the size of the model that we made at first but when the project arrived at the PCB designing part, our motive extended to space management.
+
+![workflow](https://github.com/crooknish/Smartwatch/blob/de03d77899220ad73e6d6c49e93258564fbb5392/Images%20and%20Videos/Workflow.jpeg)
+
+## Electronic aspects of the design
```
Comment (theobscuredev) [inline README.md]: ~~Electronic aspects of the design~~ -> 'Electronic Aspect of the Design'
Make sure all the headings follow the specified convention.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,140 @@
...
+<p align="center">
+  <img src="https://github.com/crooknish/Smartwatch/blob/b9872868b3e2b396aa44f89e3450bc2ef973ca1a/Images%20and%20Videos/PCB%203D%20view.jpeg"></br>
+  <i> PCB 3D Model </i>
+  </p>
+  
+
+## Motivation
+This project mainly aimed to create a rough model of a smartwatch that will resemble a typical smartwatch in functioning but, we however optimized the design so that it will also resemble a typical smartwatch in appearance too. However, the ultimate motive of this project was to learn the coding of the Arduino and testing it along with different sensors and modules that are connected to the Arduino. So, we were least bothered about the size of the model that we made at first but when the project arrived at the PCB designing part, our motive extended to space management.
+
+![workflow](https://github.com/crooknish/Smartwatch/blob/de03d77899220ad73e6d6c49e93258564fbb5392/Images%20and%20Videos/Workflow.jpeg)
```
Comment (theobscuredev) [inline README.md]: Mention a title for this image, what it depicts?

File: README.md [hunk 6]
```diff
@@ -1,2 +1,140 @@
...
+   - It consumes less power as compared to wi-fi module
+
+#### 	Buzzer
+   - Used to notify us about messages and alarms.
+   - Codable and cheap, with less power consumption.
+
+#### 	Push Buttons
+   - Used to power up and break the circuit enabling to switch between different Menu Bar
+
+![shematic in proteus](https://github.com/crooknish/Smartwatch/blob/de03d77899220ad73e6d6c49e93258564fbb5392/Images%20and%20Videos/smartwach_proteus_simulation.JPG)
```
Comment (theobscuredev) [inline README.md]: Include a title for the image.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,140 @@
...
+
+
+#### Secondary Battery - LiPo/Li-ion battery
+    •	High Cell Voltage (3.7V under Load)
+    •	Lightweight, High power Density
+    •	Output capability better than other (like- LEAD TYPE, NiMH, etc.)
+    •	Higher power to size ratio
+    
+## Mechanical aspects of the design
+For the smart watch, we made a case to protect it from external damage from scratches, any accidental drops or damage to its hardware by any means. It also makes it look good. As we are limited to work in this online environment, we have designed a virtual case in **Solidworks**.
```
Comment (theobscuredev) [inline README.md]: 'Mechanical Aspect of the Design'
Also, as specified in the guidelines, include the mechanical aspect before the electronic ones.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,140 @@
...
+
+#### Secondary Battery - LiPo/Li-ion battery
+    •	High Cell Voltage (3.7V under Load)
+    •	Lightweight, High power Density
+    •	Output capability better than other (like- LEAD TYPE, NiMH, etc.)
+    •	Higher power to size ratio
+    
+## Mechanical aspects of the design
+For the smart watch, we made a case to protect it from external damage from scratches, any accidental drops or damage to its hardware by any means. It also makes it look good. As we are limited to work in this online environment, we have designed a virtual case in **Solidworks**.   
+The case is a 51mm x 51mm x 20mm cuboid made of PVC. It is designed such that it has it has holes for the OLED pins to be attached as well as for the keys to be placed. Also, it has base which is also made of PVC and can be fastened with the case using screws.
```
Comment (theobscuredev) [inline README.md]: Include a line break before this statement.

File: README.md [hunk 9]
```diff
@@ -1,2 +1,140 @@
...
+| 3D printed Casing  | 200  |
+| Net Expected Cost  | 3090  |
+
+## Applications
+1. This smart watch functions just like other smart watches in the market. It shows us time, connects with the phone and alerts us if we receive notifications of any kind (call, messages, Whatsapp etc.).
+2. Another practical application is it measures the temperature of the surroundings where we are in. This could be useful for a lot of purposes. 
+3. Another common feature is alarm. Alarms form an integral part of our life. They act as reminders, wake us in the morning or when we take a break/nap during our routine etc. The buzzer in the smartwatch does this purpose.
+
+## Limitations
+1. 1.	One of the main limitations of the project is about the simulation environment. We didn’t have all the required features and components for simulating all the required components in a single software. This has made us to look for different alternatives like initially we started with tinkercad, then moved to Proteus 8 for simulating the hardware part of the smartwatch. For PCB designing, as we didn’t find all the required components in Proteus 8, we had to depend on EasyEDA for PCB designing.
```
Comment (theobscuredev) [inline README.md]: Remove the extra bullet point.

File: README.md [hunk 10]
```diff
@@ -1,2 +1,140 @@
...
+
+## Future Improvement  
+1. We can make our own processing board in place of arduino nano by using ATMEGA-328P processor. This will reduce the cost and space occupied.
+2. A feature like finding phone through our smartwatch can be added to it. Our smartwatch will send instructions to start ringing the phone so that we can find it easily.
+
+## Team members
+1. Abhijeet Reddy Dasari
+2. Nishant kumar
+3. Diwahar H
+4. Karan Saharan
```
Comment (theobscuredev) [inline README.md]: Hyperlink the Github ID of each team member & your mentor (https://github.com/SanjeevKrishnan). Refer this README for the same: https://raw.githubusercontent.com/marsiitr/Rube-Goldberg-Machine/master/README.md

File: README.md [hunk 11]
```diff
@@ -1,2 +1,140 @@
...
+
+## Mentor
+1. Sanjeev Krishnan R
+
+## References
+  - [solidworks](	https://www.solidworks.com/)
```
Comment (theobscuredev) [inline README.md]: Update: ~~solidworks~~ -> 'SolidWorks'

File: README.md [hunk 12]
```diff
@@ -1,2 +1,140 @@
...
+    •	High Cell Voltage (3.7V under Load)
+    •	Lightweight, High power Density
+    •	Output capability better than other (like- LEAD TYPE, NiMH, etc.)
+    •	Higher power to size ratio
+    
+## Mechanical aspects of the design
+For the smart watch, we made a case to protect it from external damage from scratches, any accidental drops or damage to its hardware by any means. It also makes it look good. As we are limited to work in this online environment, we have designed a virtual case in **Solidworks**.   
+The case is a 51mm x 51mm x 20mm cuboid made of PVC. It is designed such that it has it has holes for the OLED pins to be attached as well as for the keys to be placed. Also, it has base which is also made of PVC and can be fastened with the case using screws. 
+
+![casing img](https://github.com/crooknish/Smartwatch/blob/3f2100be9ce8981b57f7867797f3ece374e26aba/Images%20and%20Videos/case.jpeg)
```
Comment (theobscuredev) [inline README.md]: Center align the image and add a title for the same.

File: Images and Videos/smartwatch simulation video.md [hunk 13]
```diff
@@ -0,0 +1,2 @@
+Simulation video drive link -- https://drive.google.com/file/d/1OniWRCO3uQOtftLiYueDCmpmfW4u2sBm/view?usp=sharing
+Alarm simulation video drive link -- https://drive.google.com/file/d/12NadT2DNgS3l1XpdMgqY8CFpP3HzDieJ/view?usp=drivesdk
```
Comment (theobscuredev) [inline Images and Videos/smartwatch simulation video.md]: Insert a line break between these 2 statements.

File: src/smartwatch_arduino_nano.ino [hunk 14]
```diff
@@ -0,0 +1,387 @@
...
+#include <DallasTemperature.h>
+
+
+RTC_DS3231 rtc;
+char daysOfTheWeek[7][12] = {"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"};
+
+#define SCREEN_WIDTH 128   
+#define SCREEN_HEIGHT 64 
+
+#define ONE_WIRE_BUS 8
```
Comment (theobscuredev) [inline src/smartwatch_arduino_nano.ino]: Remove unnecessary line breaks & group related things together.

File: src/smartwatch_arduino_nano.ino [hunk 15]
```diff
@@ -0,0 +1,387 @@
...
+{
+  // for rectangle
+  display.clearDisplay();
+  display.setTextSize(1);
+  display.setTextColor(BLACK,WHITE);
+  display.setCursor(20,0);
+  display.println("MARS Smartwatch");
+  display.drawRoundRect(2, 12, 126, 52, 8, WHITE);
+
+  //delay(2000);
```
Comment (theobscuredev) [inline src/smartwatch_arduino_nano.ino]: Remove such debugging statements.

**Review by theobscuredev — CHANGES_REQUESTED** [review]
Hey @crooknish & team, nice work!
Although make sure you do the following suggested changes before we proceed with the merging:

1. Update the README file according to the review given for the same. Also, try to make the images in the README look smaller by setting appropriate dimensions to the same.
2. Create separate 'Images' and 'Videos' folders inside the 'Images and Videos' directory and insert the corresponding images & videos in their respective folders. Also, try to upload the videos in .mp4 format. 
3. Remove any image/video from the 'Mechanical Design' folder. It should only contain SW or corresponding file. Insert the images in the 'Images and Videos' folder.
4. Review the Arduino code once again and remove any unnecessary line breaks, spaces and importantly any debugging statements, a few of which have been marked below.

Lemme know once you've made these changes by commenting in this thread.

**Comment by crooknish** [comment]
@theobscuredev Please review the changes made and suggest if any other improvement required.

**Comment by theobscuredev** [comment]
Nice work!
@SanjeevKrishnan I suggest for merge

**Review by SanjeevKrishnan — CHANGES_REQUESTED** [review]
A very minor change is required
Rename the "Mechanical Design" folder as "Design"

**Comment by crooknish** [comment]
@SanjeevKrishnan Done the required change. You can review it once again.

File: README.md [hunk 16]
```diff
@@ -1,2 +1,174 @@
...
+#### Secondary Battery - LiPo/Li-ion battery
+
+   -	High Cell Voltage (3.7V under Load)
+   -	Lightweight, High power Density
+   -	Output capability better than other (like- LEAD TYPE, NiMH, etc.)
+   -	Higher power to size ratio
+   
+## How to simulate smartwatch in proteus
+  - Install proteus and arduino IDE and all the required libraries.
+  - Make the arrangement of all component according to the schematic.
```
Comment (SanjeevKrishnan) [inline README.md]: Step 2 has to be rewritten. Make it "Download the proteus file from [Design/Proteus](https://github.com/crooknish/Smartwatch/tree/main/Design/Proteus)"

File: README.md [hunk 17]
```diff
@@ -1,2 +1,174 @@
...
+  - Copy the code kept in src folder to the arduino and compile it.
```
Comment (SanjeevKrishnan) [inline README.md]: Point 3 has to be rewritten. Make it "Download the Arduino code from [src](https://github.com/crooknish/Smartwatch/tree/main/src) and compile it in Arduino IDE"

File: README.md [hunk 18]
```diff
@@ -1,2 +1,174 @@
...
+</p>
+
+#### Secondary Battery - LiPo/Li-ion battery
+
+   -	High Cell Voltage (3.7V under Load)
+   -	Lightweight, High power Density
+   -	Output capability better than other (like- LEAD TYPE, NiMH, etc.)
+   -	Higher power to size ratio
+   
+## How to simulate smartwatch in proteus
```
Comment (SanjeevKrishnan) [inline README.md:116]: Steps for simulation should be numbered. Don't use bullet points.

**Review by SanjeevKrishnan — CHANGES_REQUESTED** [review]
Very few final minor changes are required!

**Comment by crooknish** [comment]
@SanjeevKrishnan Made the required changes. Suggest me if any other improvement is required.

**Review by SanjeevKrishnan — APPROVED** [review]
Great Work Team! PR looks perfect! Merging the PR

**Pull request merged**